### ISSUE TITLE
How to use a triple dataset split for a calibrated classifier [ENH]: 

### ISSUE THREAD

**Issue by npapan69** [pycaret/pycaret#2890]
### Describe the feature you want to add to this project

For risk assessment models, calibration of the predicted probabilities is as important as having good discrimination between classes. When using CalibrationClassifierCV, with a base estimator that has already been fitted, the calibration should be done on a separate dataset (calibration set) from the training to avoid overfitting. At the moment, PyCaret offers only a two-dataset split (Train, Test) defined in the setup function either through a split or by explicit declaration.  

### Describe your proposed solution

As Antoni Baum kindly suggested, that may be accomplished by combining the train and calibration data before the setup function and setting as a CV splitter with correct indices using the predefined split (https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.PredefinedSplit.html#sklearn.model_selection.PredefinedSplit)

Suppose that you have three datafranes (df_train, df_cal and df_test)

#Initially, you merge the train and calibration sets.

from sklearn.model_selection import PredefinedSplit
df = df_train.merge(df_cal)

#Then define the input data for all three; GroundTruth is your target variable

X_train = df_train.drop('GroundTruth', axis=1)
X_cal = df_cal.drop('GroundTruth', axis=1)
X_test = df_test.drop('GroundTruth', axis=1)

#And then define the indices and the PredefinedSplit
train_val_idxs = np.append(np.full(X_train.shape[0], -1), np.full(X_cal.shape[0], 0))
ps = PredefinedSplit(train_val_idxs)
result = next(ps.split(df), None)

and in the setup function, you pass fold_strategy=ps

After completion of the setup function, and to get access to X_train, X_cal and X_test, you run the following code:

X_train_old = get_config('X_train')
Y_train_old = get_config('y_train')
X_test = get_config('X_test')
Y_test = get_config('y_test')

X_train = X_train_old.iloc[result[0]]
Y_train = Y_train_old.iloc[result[0]]
X_cal =  X_train_old.iloc[result[1]]
Y_cal =  Y_train_old.iloc[result[1]]

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
